tests/e2e_test/utils_test.go

```diff
@@ -202,3 +291,67 @@ func GetKymaStateMetricCount(ctx context.Context, kymaName, state string) (int,
 
 	return 0, nil
 }
+
+func UpdateSampleCRSpec(ctx context.Context, name, namespace, resourceFilePath string, clnt client.Client) error {
+	sampleCR := &unstructured.Unstructured{Object: map[string]interface{}{
+		"apiVersion": "operator.kyma-project.io/v1alpha1",
+		"kind":       "Sample",
+	}}
+
+	if err := clnt.Get(ctx, client.ObjectKey{Namespace: namespace, Name: name}, sampleCR); err != nil {
+		return err
+	}
+
+	if err := unstructured.SetNestedField(sampleCR.Object, resourceFilePath, "spec", "ResourceFilePath"); err != nil {
+		return err
+	}
+
+	return clnt.Update(ctx, sampleCR)
+}
+
+func CheckSampleCRIsInState(ctx context.Context, name, namespace string, clnt client.Client,
+	expectedState string,
+) error {
+	sampleCR := &unstructured.Unstructured{Object: map[string]interface{}{
+		"apiVersion": "operator.kyma-project.io/v1alpha1",
+		"kind":       "Sample",
+	}}
+
+	if err := clnt.Get(ctx,
+		client.ObjectKey{Name: name, Namespace: namespace}, sampleCR); err != nil {
+		return err
+	}
+
+	stateFromCR, stateExists, err := unstructured.NestedString(sampleCR.Object, "status", "status")
+	if err != nil || !stateExists {
+		return err
+	}
+
+	if stateFromCR != expectedState {
+		return fmt.Errorf("%w: expect %s, but in %s",
+			errSampleCrNotInExpectedState, expectedState, stateFromCR)
+	}
+	return nil
+}
+
+func SampleCRNoDeletionTimeStampSet(ctx context.Context, name, namespace string, clnt client.Client) error {
```

Comment:
Please consider 
1. refactor this as a more generic function to something like `CRExistsWithoutDeletionTimestamp`, this can be reused in other places.
2. or extract internal function body as a generic one and put it under `testutil` pkg, which accepts all object, not only SampleCR.

`CheckIfExists` can be reuse here, then only need to additionally check if deletionTimestamp exists.

Reply:
`ManifestNoDeletionTimeStampSet` can also share this code, what you need to do is convert typed manifest to unstructured use `runtime.DefaultUnstructuredConverter.ToUnstructured`

Reply:
Refactored it, but I disagree with `ManifestNoDeletionTimeStampSet`, since we never know the exact name of the Manifest CR. The `ManifestNoDeletionTimeStampSet ` list all manifest and gets the correct one based on the naming schema (contains kymaName and moduleName). Then we could fetch the manifest again, becaise then we have the name and namespace or we just check the DeletionTimeStamp on the already fetched object from the manifestlist, since this includes the deletiontimestamp.

Reply:
First, we can get manifest CR name from `kyma.status.modules.manifests`, you can reuse this [`GetManifest`](https://github.com/kyma-project/lifecycle-manager/pull/911/files#diff-fe4282c0e1f2cefc83a319710b3e8db61f6f40a3b927d946357e786eed6c2e6eR32) after it merged.

Second, this does not prevent you from reusing `CRExistsWithoutDeletionTimestamp`, after you get a manifest CR, no matter which way mentioned above, what you need to do is just convert it to unstructured, then you use shared code.

pseudo code:
```
ManifestNoDeletionTimeStampSet() {
 manifest, err := getmanifest();
 manifestUnstructured := runtime.DefaultUnstructuredConverter.ToUnstructured(manifest)
 CRExistsWithoutDeletionTimestamp(manifestUnstructured)
}
```